src/main/java/com/springboot/star_wars_character_popularity/app/service/serviceImpl/UserServiceImpl.java

```diff
@@ -1,38 +1,52 @@
 package com.springboot.star_wars_character_popularity.app.service.serviceImpl;
 
 import com.springboot.star_wars_character_popularity.app.exception.ResourceNotFoundException;
-import com.springboot.star_wars_character_popularity.app.model.RoleModel;
-import com.springboot.star_wars_character_popularity.app.model.UserModel;
+import com.springboot.star_wars_character_popularity.app.model.Role;
+import com.springboot.star_wars_character_popularity.app.model.User;
 import com.springboot.star_wars_character_popularity.app.repository.UserRepository;
 import com.springboot.star_wars_character_popularity.app.service.UserService;
-import org.apache.catalina.User;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
 import org.springframework.stereotype.Service;
 
+import java.util.Collection;
 import java.util.List;
 import java.util.Optional;
+import java.util.stream.Collectors;
 
 @Service
 public class UserServiceImpl implements UserService {
 
+    @Autowired
+    private BCryptPasswordEncoder passwordEncoder;
+
     private UserRepository userRepository;
 
     public UserServiceImpl(UserRepository userRepository){
         this.userRepository=userRepository;
     }
 
     @Override
-    public UserModel saveUser(UserModel user) {
+    public User saveUser(User user) {
+
```

Comment:
only regular Users can be created. Maybe it's a good idea to put his ROLE here ?